plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/configuration/DatabaseMetadata.java

```diff
@@ -147,15 +104,124 @@ private static File getDefaultMetadataFile(final Path dataDir) {
 
   private static DatabaseMetadata resolveDatabaseMetadata(final File metadataFile)
       throws IOException {
-    DatabaseMetadata databaseMetadata;
     try {
-      databaseMetadata = MAPPER.readValue(metadataFile, DatabaseMetadata.class);
+      try {
+        return tryReadAndMigrateV1(metadataFile);
+      } catch (DatabindException dbe) {
+        return tryReadV2(metadataFile);
+      }
     } catch (FileNotFoundException fnfe) {
-      databaseMetadata = new DatabaseMetadata(1, 1);
+      throw new StorageException(
```

Comment:
This is just a comment, non blocking feedback, are we planning to automate a solution for this catastrophic case in the future? Maybe it's not worth trying and users will have to wipe db and resync....

Reply:
I do not think automatic recover is worth the effort, since the lack of that file means the user did an error, and recovering from it means to try to identify the format and version querying the db, and if we implement that, then the metadata itself is redoundant